presentation/src/main/java/com/safetyheads/akademiaandroida/presentation/ui/fragments/profile/ProfileInformationFragment.kt

```diff
@@ -0,0 +1,308 @@
+package com.safetyheads.akademiaandroida.presentation.ui.fragments.profile
+
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import com.safetyheads.akademiaandroida.domain.entities.firebasefirestore.Profile
+import com.safetyheads.akademiaandroida.presentation.databinding.FragmentProfileInformationBinding
+import com.safetyheads.akademiaandroida.presentation.ui.utils.AddressValidator
+import com.safetyheads.akademiaandroida.presentation.ui.utils.CityValidator
+import com.safetyheads.akademiaandroida.presentation.ui.utils.FullNameValidator
+import com.safetyheads.akademiaandroida.presentation.ui.utils.PhoneNumberValidator
+import com.safetyheads.akademiaandroida.presentation.ui.viewmodels.ProfileViewModel
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import org.koin.androidx.viewmodel.ext.android.activityViewModel
+
+class ProfileInformationFragment : Fragment() {
+
+    private lateinit var binding: FragmentProfileInformationBinding
+    private val viewModel: ProfileViewModel by activityViewModel()
+
+    private lateinit var actualProfile: Profile
+    private var fullNameJob: Job? = null
+    private var jobPositionJob: Job? = null
+    private var phoneNumberJob: Job? = null
+    private var streetJob: Job? = null
+    private var cityJob: Job? = null
+    private var countryJob: Job? = null
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentProfileInformationBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initUI()
+        observeUI()
+    }
+
+    private fun observeUI() {
+        viewModel.userInformation.observe(viewLifecycleOwner) { profileInformation ->
+            if (profileInformation != null) {
+                actualProfile = profileInformation
+                binding.apply {
+                    if (fullNameJob == null)
+                        fullNameEditText.setText("${profileInformation.firstName} ${profileInformation.lastName}")
+                    if (jobPositionJob == null)
+                        jobPositionEditText.setText(profileInformation.jobPosition)
+                    if (phoneNumberJob == null)
+                        phoneNumberEditText.setText(profileInformation.phoneNumber)
+                    if (streetJob == null)
+                        streetEditText.setText("${profileInformation.address.streetName} ${profileInformation.address.streetNumber}")
```

Comment:
I think you can split it into two edit texts: one with street name and second with street number